cpp/src/arrow/chunk_resolver.h

```diff
@@ -36,8 +39,42 @@ struct ChunkLocation {
 
   /// \brief Index of the value in the chunk
   ///
-  /// The value is undefined if chunk_index >= chunks.size()
+  /// The value is UNDEFINED if chunk_index >= chunks.size()
   int64_t index_in_chunk = 0;
+
+  /// \brief Create a ChunkLocation without asserting any preconditions.
+  static ChunkLocation Forge(int64_t chunk_index, int64_t index_in_chunk) {
+    ChunkLocation loc;
+    loc.chunk_index = chunk_index;
+    loc.index_in_chunk = index_in_chunk;
+    return loc;
+  }
+
+  ChunkLocation() = default;
+
+ public:
```

Comment:
The above are also public as this is a `struct`.

Reply:
I didn't mean for the others to be `private` fields. While iterating on this I had a `private:` section that I ended up removing.